library/src/interactor_impl.cxx

```diff
@@ -266,15 +256,30 @@ class interactor_impl::internals
         else if (keySym == "Return")
         {
           ren->ResetCamera();
-          renWin->Render();
         }
         else if (keySym == "Space")
         {
           self->AnimationManager.ToggleAnimation();
-          renWin->Render();
         }
         break;
     }
+
+    if (checkColoring)
+    {
+      // TODO This is not super great, do better ?
+      bool useCell;
+      std::string arrayName;
+      int component;
+      renWithColor->GetColoring(useCell, arrayName, component);
+      self->Options.set("cells", useCell);
+      self->Options.set("scalars", arrayName);
+      self->Options.set("component", component);
```

Comment:
Not sure to follow, but I guess this implementation is fine.

Reply:
The main difference between coloring and the other options is that instead of changing the options and letter the vtkF3DRenderer handle it, we control the renderer directly and modify the options after to make it match.